### ISSUE TITLE
Windows + any source. 

### ISSUE THREAD

**Issue by ReneFroger** [junkblocker/codesearch-pre-github#5]
Thanks for sharing your fork with us! 

I really would like to use this fork for Unite codesearch. 

Due to work requirements, I'm currently bounded to Windows. The Codesearch binaries for Windows are working fine (e.g. they are stored in my path, when I do `csearch test`, they're working fine. 

It seems they're not working together with your Unite codesearch. 

Is there any chance to get this compiled into a binary, in order to get your plugin working for for Windows? Thanks in advance.

**Comment by junkblocker** [comment]
hi @ReneFroger ,  I am not sure I understand what you mean by getting it 'compiled into a binary'. You already mention that you have a `csearch` binary that works from the command line.

You can try setting `g:unite_codesearch_command` to specify the complete path of the `csearch` executable on your system just in case Vim isn't picking it up for some reason. This has to look something like

```
  let s:codesearch_command = "c:\\your\\csearch\\bin\\dir\\csearch.exe %d "%s"'
```

or maybe

```
  let s:codesearch_command = "c:/your/csearch/bin/dir/csearch.exe %d "%s"'
```

. Let me know if that helps.

**Comment by ReneFroger** [comment]
Hi @junkblocker, thanks for your kind reply. Are you sure that the first example path is correct? Because the `"%s"'` escaped the first double quote. So guess you mean `'%s'` and then double quote? 

I tried both path with the really path to my csearch.exe, with the %s between single quotes. Then `:Unite codesearch` without any result. I don't get any error message or something, the screen stays empty. 

In case of above is unclear to you, say it please. Then I will record a screencast of the progress. 

Any tips by the chance?

**Comment by junkblocker** [comment]
Sorry, that was just a bunch of typos. I meant to have single quotes outside and double quotes around `%s` (and also -m) like so:

```
  let s:codesearch_command = 'c:/your/csearch/bin/dir/csearch.exe -m %d "%s"'
```

Does running

```
:!csearch -m 10 i
```

(or any other variations with `csearch.exe`, full path etc. in Vim produce any output for you?

**Comment by ReneFroger** [comment]
Thanks for your reply, yes I was already saying that. I tried that already with the single quotes. :)
I tried the following: 

```
let s:codesearch_command = "C:\\Dropbox\\Vim\\vim74\\csearch.exe -m %d '%s'"
```

And 

```
let s:codesearch_command = "C:/Dropbox/Vim/vim74/csearch.exe -m %d '%s'"
```

With `:Unite codesearch`, I get no results. I noticed the input is invalide with the red label. But I don't see any error message whatsoever. 

In my Windows cmd, I run only the words `csearch exampleword` and it gives me the desired output (without -m or another tags). Any idea how I could look further for a solution? I can make a screencast to make it more clear where things went wrong on my side, if you want.

**Comment by junkblocker** [comment]
I need to know the answer to running

```
:!csearch -m 10 i
```

inside vim not in Windows cmd. That will tell me if path or related issues are occuring. While we are at it, you could also tell me if (again inside vim) just

```
:!csearch
```

produces the help output.

**Comment by ReneFroger** [comment]
Sorry for my belated response, due to work load. But here is my reply then! :-)

When I do `:!csearch -m 10 i`, I get this output: 

![schermafdruk 2014-09-23 22 21 25](https://cloud.githubusercontent.com/assets/5211845/4379307/660e5080-435f-11e4-8d0a-26c8f9ae68f3.png)

Then `:!csearch` gives me this:

![schermafdruk 2014-09-23 22 21 51](https://cloud.githubusercontent.com/assets/5211845/4379310/6e5ae334-435f-11e4-8529-71937a07904c.png)

It's responding, it seems? But when I do csearch exampleword-to-search, it will find. Is that output useful to be wrapped into your plugin? 
(I wanted this feature, because I'm planning to make a video about how useful Unite is, which it will be displayed on shougo/unite github page, and on reddit.com/r/vim). 

Thanks in advance for your reply!

**Comment by junkblocker** [comment]
This looks like you are using a different codesearch (maybe from the original google version?) and not my this fork of codesearch. I have added the `-m` option to my fork to limit the amount of results returned. So please try with this codesearch and let me know if it still does not work.

**Comment by ReneFroger** [comment]
Yes, I use the original Google version. It was the only Windows binary of Codesearch that I could find.  I never thought that your fork would be working on Windows. As it seems, I couldn't find any binaries in your fork. Only go-files. I. Will try to figure out the Go language and how to get working on Windows, use your fork, and then reply you back. Sorry for any inconvience.

**Comment by junkblocker** [comment]
No worries. If you install go ( http://golang.org/dl/ ), you can just do

```
go get github.com/junkblocker/codesearch/...
```

to get my fork built for windows acc. to another user I've talked to in the past. Let me know how that goes.

I don't really want to get into providing binaries unless go's cross platform builds also work for windows. That's something I have on my list of things to explore sometime.

**Comment by junkblocker** [comment]
So go does do cross platform builds pretty easily and I was able to build the windows binaries. However I want to automate the process of publishing the binaries here, figure out my tagging scheme and the release strategy. So, that will need some time so it will probably need to wait for this weekend.

Meanwhile, if you want the binaries I build, let me know an email address to send it to. Mine is in my profile. Specify which platforms (windows, linux etc.) and archs (386, amd64) etc. you want.

**Comment by ReneFroger** [comment]
Awesome! You're my new hero! :-)

I use Windows 7 x64 with amd64 chipset. Is that possible to compile anything in that instruct set, if I might ask? I will e-mail you now.
When the codesearch feature is working, I will record my video and share it!

**Comment by junkblocker** [comment]
:) Yes, go can cross compile for `386`, `amd64` and some `arm` for most all OSes. Sent you the download link by email.

**Comment by ReneFroger** [comment]
Thanks for your mail, I continue the conversation here for future reference for others. I grabbed the binaries from your downloadlink, and replaced csearch.exe, cgrep.exe and cindex.exe in the folder vim74.

When I do run a search, it worked fine:

![schermafdruk 2014-09-24 22 42 49](https://cloud.githubusercontent.com/assets/5211845/4395483/638cb500-442b-11e4-8ecb-a4693d11455b.png)

After starting up Vim, I run `:Unite codesearch`. It's already marked red, even when I don't typed anything yet, don't know why. I can type words, but they're still marked red and don't get any results after Enter (yes, I tried hlsearch too)

![schermafdruk 2014-09-24 22 46 46](https://cloud.githubusercontent.com/assets/5211845/4395532/ec93dde2-442b-11e4-9267-c1ad8bfebb8b.png)

In my Vimrc settings, I tried the following:

```
let s:codesearch_command = "C:\\Dropbox\\Vim\\vim74\\csearch.exe" 
```

and 

```
let s:codesearch_command = "C:\\Dropbox\\Vim\\vim74\\csearch.exe -m %d '%s'"
```

I tried it with single backslash too, without result. Any tips where I could get looking? In the command line it worked fine.

**Comment by junkblocker** [comment]
It's missing the `-m %d "%s"` part. Try with 

```
let s:codesearch_command = "C:\\Dropbox\\Vim\\vim74\\csearch.exe -m %d '%s'"
```

or

```
let s:codesearch_command = "C:/Dropbox/Vim/vim74/csearch.exe -m %d '%s'"
```

etc.

**Comment by ReneFroger** [comment]
Both lines didn't worked yesterday. I tried with another variant

```
let s:codesearch_command = "C:\Dropbox\Vim\vim74\csearch.exe -m %d '%s'"
```

Without result. Any tips how I could debug this?

**Comment by junkblocker** [comment]
Silly me, copy paste issue :( The variable needs to be `g:unite_codesearch_command` and not `s:codesearch_command` because that's only for script use. Sorry for so much waste of time :(

**Comment by ReneFroger** [comment]
No problem! Already happy that you would help me. Well, it changed something.

With both

```
let g:unite_codesearch_command = "C:\DropboxVim\vim74\csearch.exe -m %d '%s'" 
let g:unite_codesearch_command = "C:\\Dropbox\\Vim\\vim74\\csearch.exe -m %d '%s'"
```

I get error messages when I type my input into Unite codesearch, Propably from Vimproc that he couldnt find the Windows style path. With the POSIX path,

```
let g:unite_codesearch_command = "C:/Dropbox/Vim/vim74/csearch.exe -m %d '%s'"
```

I don't get any error. Propably Vim interpret the right path there. But whatever I type, like hlsearch, the keys are invalid (as in, red label from Unite, that he couldn't find anything, like in the screenshot that I posted earlier).

Then tested with csearch in cmdprompt, it worked fine. Any tips?

**Comment by junkblocker** [comment]
Maybe you are not starting in insert mode. So try

```
:Unite -start-insert codesearch
```

Beyond that it'd be hard to figure out without actually being on the system.

**Comment by ReneFroger** [comment]
Thanks for your reply, didn'thelp. But I was thinking about this issue,  I will mail you tommorow with another possible way to finding a solution to this issue. Thanks for your time, by the way!

**Comment by junkblocker** [comment]
I've done a [binaries release](https://github.com/junkblocker/codesearch/releases). I'm closing this bug now as the remaining part is really related to unite-codesearch on windows.

**Comment by ReneFroger** [comment]
However the Unite codesearch never worked on Windows yet, [I mentioned Unite Codesearch in video](http://www.reddit.com/r/vim/comments/2j2f9p/video_navigate_with_unite_especially_for_users_of).

Maybe it get some traction in the Vim community that it deserved!

**Comment by junkblocker** [comment]
@ReneFroger, sorry I got busy and didn't reply to your email. Thanks for the mention, :thumbsup:  ! I have to say I saw the video and my Mac/FreeBSD selves are hurt at seeing 'Linux only', :broken_heart:  :stuck_out_tongue_winking_eye: . I realize you were speaking in layman terms :smile: .

I've moved this issue over to junkblocker/unite-codesearch#1 and have an idea what's happening. Let's continue the discussion over there.
